node/integration_test.go

```diff
@@ -162,12 +163,107 @@ func TestTxGossipingAndAggregation(t *testing.T) {
 			require.NoError(err)
 			nodeBlock, err := nodes[i].Store.LoadBlock(h)
 			require.NoError(err)
+			// Only Intermediate state roots set by block aggregator are relevant, removed for sake of comparison
+			nodeBlock.Data.IntermediateStateRoots.RawRootsList = nil
 			assert.Equal(aggBlock, nodeBlock)
 		}
 	}
 }
 
-func createNodes(aggCtx, ctx context.Context, num int, wg *sync.WaitGroup, t *testing.T) ([]*Node, []*mocks.Application) {
+func TestFraudProofTrigger(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	var wg sync.WaitGroup
+	clientNodes := 4
+	aggCtx, aggCancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
+	nodes, apps := createNodes(aggCtx, ctx, clientNodes+1, true, &wg, t)
+
+	wg.Add((clientNodes + 1) * clientNodes)
+	for _, n := range nodes {
+		require.NoError(n.Start())
+	}
+
+	// wait for nodes to start up and establish connections; 1 second ensures that test pass even on CI.
+	time.Sleep(1 * time.Second)
+
+	for i := 1; i < len(nodes); i++ {
+		data := strconv.Itoa(i) + time.Now().String()
+		require.NoError(nodes[i].P2P.GossipTx(context.TODO(), []byte(data)))
+	}
+
+	timeout := time.NewTimer(time.Second * 30)
```

Comment:
why 30s?

Reply:
This is a safety limit to ensure test will end reasonably soon. 30s is a very safe, big timeout, and it's motivated by previous issues with slow CI.